src/main/java/de/dhbw/karlsruhe/top/down/parsing/validation/TopDownAcceptorValidation.java

```diff
@@ -162,6 +162,10 @@ private boolean validateProductionStep(TopDownStep tdStep, TopDownStep nextStep)
     private boolean checkUsedProduction(String stackCurrStep, String stackNextStep, GrammarProduction productionCurrStep) {
         boolean success = false;
 
+        if (!this.grammarService.getGrammarRules().contains(productionCurrStep)) {
```

Comment:
Vielleicht noch einen TestCase hierfür hinzufügen. Also einen TopDownAcceptor welcher an sich richtig (richtige Schritte, richtiges Wort, etc.) ist aber eine falsche Produtkion nutzt, welche nicht in der angegebene Grammatik ist.